presto-main/src/main/java/com/facebook/presto/operator/LookupJoinOperator.java

```diff
@@ -233,21 +226,41 @@ private Page mapPage(IntArrayList unspilledPositions, Page page)
     @Override
     public Page getOutput()
     {
-        if (finishing && spiller.isPresent()) {
-            checkState(spilledLookupJoiner.isPresent());
+        if (finishing && spiller.isPresent() && currentLookupPartition.map(CompletableFuture::isDone).orElse(false)) {
```

Comment:
Maybe put `checkState(!currentLookupPartition.isPresent() || currentLookupPartition.get().isDone())` before this if?